### PR TITLE
[Ready for Review] Set up testing for views

### PR DESCRIPTION
**Pull request by helendduncan**
Test for model (basic str test only) and views (share_exp) currently only with placeholder descriptive titles. 
Addresses a task is issue #461 
Around 87% coverage of views file (96% overall) - marking as ready for review as risks going stale again

### CODE REVIEW

**Comment by gedankenstuecke** [comment]
I guess this PR might have gotten a bit stale with respect to the other tests that were written (i.e. i'm not sure how up2date the CI setup of this one is)? 

@helendduncan Do you reckon it might be easier to just copy over the model tests into a new PR instead of updating this one? :)

**Comment by helendduncan** [comment]
Yeah I've been wanting to chat about the API "calls" perhaps you are free at some point Wednesday?

**Comment by gedankenstuecke** [comment]
Yes, of course! Would at some point during the morning work? 😄

**Comment by helendduncan** [comment]
I'm free between 10:00 and 12:30

**Comment by helendduncan** [comment]
coverage of 87.82 % of current view file. Placing as ready to review as I think that is high enough to be working with and am conscious of changes being made to some of the search functionality in one of my current PRs

**Review by gedankenstuecke — APPROVED** [review]
That's some amazing work @helendduncan, really well done and I think this'll overall boost our test coverage quite a bit, which should help make us feel a lot more confident for any changes we wanna implement going forward! 🎉 

I just left 2-3 minor comments. I think the assertion for the code 302 is maybe a more general one, as I think there's a few more tests where we only assert a redirect but not to where. For those cases I recommend having a look at the `assertRedirects` function (cf. in the tests for moderation views here: https://github.com/alan-turing-institute/AutSPACEs/blob/a01c22e561212b7fb31c1b2392df15e424029b05/server/apps/main/tests/test_moderate_views.py#L62) 

That function allows testing for "type of redirect", status-code of final page, and target of redirect :)

File: server/apps/main/tests/test_views.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    def test_logout_user(self):
+        """
+        Test that a user can log themselves out
+        """
+        c = Client()
+        c.force_login(self.user_a)
+        logged_in_response = c.get('/main/share_exp/')
+        assert logged_in_response.status_code == 200
+        logged_out_response = c.post("/main/logout/")
+        assert logged_out_response.status_code == 302
```
[the remark this answers is not in the document]
Comment (helendduncan) [reply server/apps/main/tests/test_views.py]: Yep done!

File: server/apps/main/tests/test_views.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+                      record_mode='none',
+                      filter_query_parameters=['access_token'],
+                      match_on=['path'])
+    def test_view_exp_logged_in(self):
+        """
+        Test that a user can view their experience when logged in and gets redirected to overview when not
+        """
+        c = Client()
+        unauthorised_response = c.get('/main/view/33b30e22-f950-11ed-8488-0242ac140003/')
+        assert unauthorised_response.status_code == 302
```
[the remark this answers is not in the document]
Comment (helendduncan) [reply server/apps/main/tests/test_views.py:145]: Test now checks redirect goes to home as expected

File: server/apps/main/tests/test_views.py [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+        self.oh_a = OpenHumansMember.create(oh_id=12345678,data=data)
+        self.oh_a.save()
+        self.user_a = self.oh_a.user
+        self.user_a.openhumansmember = self.oh_a
+        self.user_a.set_password('password_a')
+        self.user_a.save()
+        # Create user B
+        self.oh_b = OpenHumansMember.create(oh_id=87654321,data=data)
+        self.oh_b.save()
+        self.user_b = self.oh_a.user
```
[the remark this answers is not in the document]
Comment (helendduncan) [reply server/apps/main/tests/test_views.py]: Yes it should be!

**Comment by gedankenstuecke** [comment]
Thanks so much @helendduncan, amazing work on this 🙏🙂

**Pull request merged**